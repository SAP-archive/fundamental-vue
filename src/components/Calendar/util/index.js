export const weekFromDate = (
  date,
  { firstDayOfWeek } = { firstDayOfWeek: 0 }
) => {
  const startOfWeek = new Date(date.valueOf());
  startOfWeek.setDate(date.getDate() - date.getDay() + firstDayOfWeek);

  const endOfWeek = new Date(date.valueOf());
  endOfWeek.setDate(date.getDate() + (7 - firstDayOfWeek - date.getDay()));

  const days = [];
  let dayOfWeek = 0;
  while (dayOfWeek < 7) {
    const dayInWeek = new Date(startOfWeek.valueOf());
    dayInWeek.setDate(dayInWeek.getDate() + dayOfWeek);
    days.push(dayInWeek);
    dayOfWeek++;
  }
  return days;
};

export const monthFromDate = (
  referenceDate,
  options = { firstDayOfWeek: 0 }
) => {
  const month = referenceDate.getMonth();
  const result = [];
  const firstDayOfMonth = new Date(referenceDate.getFullYear(), month, 1);
  const firstWeek = weekFromDate(firstDayOfMonth, options);
  const [startingDate] = firstWeek;
  let currentMonth = month;
  while (currentMonth === month) {
    const week = weekFromDate(startingDate, options);
    result.push(week);
    startingDate.setDate(startingDate.getDate() + 7);
    currentMonth = startingDate.getMonth();
  }
  return result;
};

export const sameYear = (lhs, rhs) => lhs.getFullYear() === rhs.getFullYear();
export const sameMonth = (lhs, rhs) =>
  sameYear(lhs, rhs) && lhs.getMonth() === rhs.getMonth();
export const sameDay = (lhs, rhs) =>
  sameMonth(lhs, rhs) && lhs.getDate() === rhs.getDate();
export const earlierDate = (lhs, rhs) =>
  lhs.getTime() < rhs.getTime() ? lhs : rhs;
export const laterDate = (lhs, rhs) =>
  lhs.getTime() < rhs.getTime() ? rhs : lhs;
