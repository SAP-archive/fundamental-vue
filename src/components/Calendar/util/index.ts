import * as Types from './types';
export type Week = Types.Week;
export type Month = Types.Month;

type Options = {
  firstDayOfWeek: number;
};

export const weekFromDate = (date: Date, { firstDayOfWeek }: Options = { firstDayOfWeek: 0 }): Week => {
  const startOfWeek = new Date(date.valueOf());
  startOfWeek.setDate(date.getDate() - date.getDay() + firstDayOfWeek);

  const endOfWeek = new Date(date.valueOf());
  endOfWeek.setDate(date.getDate() + ((7 - firstDayOfWeek) - date.getDay()));

  const days: Week = [];
  let dayOfWeek = 0;
  while(dayOfWeek < 7) {
    const dayInWeek = new Date(startOfWeek.valueOf());
    dayInWeek.setDate(dayInWeek.getDate() + dayOfWeek);
    days.push(dayInWeek);
    dayOfWeek++;
  }
  return days;
};

export const monthFromDate = (referenceDate: Date, options: Options = { firstDayOfWeek: 0 }): Month => {
  const month = referenceDate.getMonth();
  const result: Month = [];
  const firstDayOfMonth = new Date(referenceDate.getFullYear(), month, 1);
  const firstWeek = weekFromDate(firstDayOfMonth, options);
  const [startingDate] = firstWeek;
  let currentMonth = month;
  while(currentMonth === month) {
    const week = weekFromDate(startingDate, options);
    result.push(week);
    startingDate.setDate(startingDate.getDate() + 7);
    currentMonth = startingDate.getMonth();
  }
  return result;
};

export const sameYear = (lhs: Date, rhs: Date) => lhs.getFullYear() === rhs.getFullYear();
export const sameMonth = (lhs: Date, rhs: Date) => sameYear(lhs, rhs) && lhs.getMonth() === rhs.getMonth();
export const sameDay = (lhs: Date, rhs: Date) => sameMonth(lhs, rhs) && lhs.getDate() === rhs.getDate();
export const earlierDate = (lhs: Date, rhs: Date): Date => lhs.getTime() < rhs.getTime() ? lhs : rhs;
export const laterDate = (lhs: Date, rhs: Date): Date => lhs.getTime() < rhs.getTime() ? rhs : lhs;
