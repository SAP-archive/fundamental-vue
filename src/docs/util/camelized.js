/**
 * @param {string} text
 */
export default text =>
  text
    .split('-')
    .map(w => w.replace(/./, m => m.toUpperCase()))
    .join('')
