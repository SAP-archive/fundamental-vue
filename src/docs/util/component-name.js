// @ts-check
import kebabCase from "./kebab-case";
import slugified from "./slugify";
import deslugified from "./deslugify";

// This class helps us to migrate between different representations of a component name.
// There are different naming conventions.

/**
 * @param {any} invalidName
 * @returns {never}
 */
const handleInvalidName = invalidName => {
  throw Error(`Invalid component name: ${invalidName}`);
};

const NORMALIZED_PREFIX = "fd";

class ComponentName {
  /**
   * @param {any} raw
   * @returns {boolean}
   */
  static hasPrefix(raw) {
    if (raw == null) {
      return false;
    }
    if (typeof raw !== "string") {
      return false;
    }
    return raw.toLowerCase().startsWith(NORMALIZED_PREFIX);
  }
  /**
   * @param {any} raw
   * @returns {ComponentName}
   */
  static from(raw) {
    if (raw == null) {
      handleInvalidName(raw);
    }
    if (typeof raw !== "string") {
      handleInvalidName(raw);
    }

    if (raw.length <= NORMALIZED_PREFIX.length) {
      // Name has no prefix
      // let's kebab-case the raw string.
      return new ComponentName(`${NORMALIZED_PREFIX}-${kebabCase(raw)}`);
    }

    if (raw.toLowerCase().startsWith(NORMALIZED_PREFIX)) {
      return new ComponentName(kebabCase(raw));
    }
    return new ComponentName(`${NORMALIZED_PREFIX}-${kebabCase(raw)}`);
  }

  /** @param {string} normalized */
  constructor(normalized) {
    this._normalized = normalized;
  }
  /** @returns {string} */
  get normalized() {
    return this._normalized;
  }
  /** @returns {string} */
  get slugified() {
    return slugified(this.normalized);
  }
  /** @returns {string} */
  get slugifiedWithoutPrefix() {
    return slugified(this.normalizedWithoutPrefix);
  }

  /** @returns {string} */
  get normalizedWithoutPrefix() {
    return this.normalized.substring(3);
  }

  /** @returns {string} */
  get displayable() {
    return deslugified(this.normalizedWithoutPrefix);
  }
  /** @returns {string} */
  get displayableWithPrefix() {
    return this.normalized;
  }
}

export default ComponentName;
