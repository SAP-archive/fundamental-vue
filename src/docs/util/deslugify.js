// @ts-check

/** @param {string} text */
const capitalized = text => {
  if (text.length === 0) {
    return text;
  }
  const firstLetter = text.charAt(0);
  return `${firstLetter.toUpperCase()}${text.substring(1)}`;
};

/** @param {string} text */
const deslugify = text => {
  const parts = text.split("-");
  return parts.map(capitalized).join(" ");
};

export default deslugify;
