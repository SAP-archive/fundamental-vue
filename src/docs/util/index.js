export { default as objectValues } from "./objectValues";
