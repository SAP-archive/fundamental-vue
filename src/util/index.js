export const noop = () => {
  /* empty */
};
export { default as objectValues } from "./objectValues";
export { default as pluginify } from "./pluginify";
export { default as withoutDuplicates } from "./without-duplicates";
