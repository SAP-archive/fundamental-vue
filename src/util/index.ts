export { componentName } from './component-name';
export { directiveName } from './directive-name';
export { targetValue } from './target-value';
export const noop = () => { /* empty */ };
