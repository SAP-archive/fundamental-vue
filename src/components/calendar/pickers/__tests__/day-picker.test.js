import { mount } from '@vue/test-utils'
import FdDayPicker from './../day-picker.vue'
import { monthFromDate } from './../../util'
import sameDay from './../../../../util/date/same-day'
import createNormalizedDate from './../../../../util/date/normalized-date'

/**
 * @typedef {Object} CreateOptions
 * @prop {"single" | "range"} mode
 * @prop {Date} today
 * @prop {Date} from
 * @prop {Date?} to
 */

/** @param {CreateOptions} options */
const createDayPicker = options => {
  const { mode, today, from, to = null } = options

  return mount({
    components: { FdDayPicker },
    template: `
      <fd-day-picker
        :month="month"
        :day-names="dayNames"
        :mode="mode"
        :displayedMonth="displayedMonth"
        :selectionStart="selectionStart"
        :selectionEnd="selectionEnd"
        :selectionContainsDate="selectionContainsDate"
        :isPresent="isPresent"
      />
    `,
    methods: {
      isPresent(date) {
        return sameDay(this.today, date)
      },
      selectionContainsDate(date) {
        const selection = {
          from: this.selectionStart,
          to: this.selectionEnd
        }
        return createNormalizedDate(selection).contains(date)
      }
    },
    computed: {
      displayedMonth: {
        set(month) {
          const copy = new Date(this.displayedDate)
          copy.setMonth(month)
          this.displayedDate = copy
        },
        get() {
          const date = this.displayedDate
          return date.getMonth()
        }
      }
    },
    data() {
      const selectionStart = from
      const month = monthFromDate(today)
      const dayNames = ['Su', 'M', 'Tu', 'W', 'Th', 'F', 'S']
      return {
        displayedDate: today,
        today,
        month,
        dayNames,
        mode,
        selectionStart,
        selectionEnd: to
      }
    }
  })
}

describe('fd-day-picker', () => {
  it('renders single selection correctly', () => {
    const wrapper = createDayPicker({
      mode: 'single',
      today: new Date('2019-06-20'),
      from: new Date('2019-06-21')
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders multiple selection correctly', () => {
    const wrapper = createDayPicker({
      mode: 'range',
      today: new Date('2019-06-20'),
      from: new Date('2019-06-21'),
      to: new Date('2019-06-24')
    })
    expect(wrapper.element).toMatchSnapshot('range selection')
  })
})
