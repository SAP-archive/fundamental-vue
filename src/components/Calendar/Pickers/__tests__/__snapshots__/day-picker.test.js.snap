// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fd-day-picker renders multiple selection correctly: range selection 1`] = `
<div
  aria-hidden="false"
  class="fd-calendar__dates"
>
  <table
    class="fd-calendar__table"
    role="grid"
  >
    <thead
      class="fd-calendar__group"
    >
      <tr
        class="fd-calendar__row"
      >
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            Su
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            M
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            Tu
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            W
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            Th
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            F
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            S
          </span>
        </th>
      </tr>
    </thead>
     
    <tbody
      class="fd-calendar__group"
    >
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            26
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            27
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            28
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            29
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            30
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            31
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            1
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            2
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            3
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            4
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            5
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            6
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            7
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            8
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            9
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            10
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            11
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            12
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            13
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            14
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            15
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            16
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            17
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            18
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            19
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--current"
        >
          <span
            class="fd-calendar__text"
          >
            20
          </span>
        </td>
        <td
          class="fd-calendar__item is-selected is-selected-range-first"
        >
          <span
            class="fd-calendar__text"
          >
            21
          </span>
        </td>
        <td
          class="fd-calendar__item is-selected-range"
        >
          <span
            class="fd-calendar__text"
          >
            22
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item is-selected-range"
        >
          <span
            class="fd-calendar__text"
          >
            23
          </span>
        </td>
        <td
          class="fd-calendar__item is-selected is-selected-range-last"
        >
          <span
            class="fd-calendar__text"
          >
            24
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            25
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            26
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            27
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            28
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            29
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            30
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            1
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            2
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            3
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            4
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            5
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            6
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`fd-day-picker renders single selection correctly 1`] = `
<div
  aria-hidden="false"
  class="fd-calendar__dates"
>
  <table
    class="fd-calendar__table"
    role="grid"
  >
    <thead
      class="fd-calendar__group"
    >
      <tr
        class="fd-calendar__row"
      >
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            Su
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            M
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            Tu
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            W
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            Th
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            F
          </span>
        </th>
        <th
          class="fd-calendar__column-header"
        >
          <span
            class="fd-calendar__day-of-week"
          >
            S
          </span>
        </th>
      </tr>
    </thead>
     
    <tbody
      class="fd-calendar__group"
    >
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            26
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            27
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            28
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            29
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            30
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            31
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            1
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            2
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            3
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            4
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            5
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            6
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            7
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            8
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            9
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            10
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            11
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            12
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            13
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            14
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            15
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            16
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            17
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            18
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            19
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--current"
        >
          <span
            class="fd-calendar__text"
          >
            20
          </span>
        </td>
        <td
          class="fd-calendar__item is-selected"
        >
          <span
            class="fd-calendar__text"
          >
            21
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            22
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            23
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            24
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            25
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            26
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            27
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            28
          </span>
        </td>
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            29
          </span>
        </td>
      </tr>
      <tr
        class="fd-calendar__row"
      >
        <td
          class="fd-calendar__item"
        >
          <span
            class="fd-calendar__text"
          >
            30
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            1
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            2
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            3
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            4
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            5
          </span>
        </td>
        <td
          class="fd-calendar__item fd-calendar__item--other-month"
        >
          <span
            class="fd-calendar__text"
          >
            6
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
